tests/conversion/utils_test.py

```diff
@@ -0,0 +1,37 @@
+import os
+import shutil
+import tempfile
+
+import pytest
+
+from osmaxx.conversion.converters.utils import recursive_getsize
+
+
+@pytest.fixture
+def directory_with_subdirectories(request):
+    tempdir = tempfile.TemporaryDirectory()
+
+    def rm_tmpdir():
+        shutil.rmtree(tempdir.name)
+    request.addfinalizer(rm_tmpdir)
+
+    file_sizes = [102, 304, 10, 2000, 10, 678691, 1667, 1254]
+    expected_size = sum(file_sizes)
+    subdir = tempdir.name
+    for index, file_size in enumerate(file_sizes):
+        if index % 2 == 0:
+            subdir = os.path.join(subdir, str(file_size))
```

Comment:
It's not "hard" _per se_. I just had to read this loop very carefully to figure out what it's doing (all the while guessing why it's doing what it's doing), whereas if the loop was
```python
for dir_path, filename, file_size in files_to_create_with_sizes:
    os.mkdir(path, exists_ok=True)
    file_path = os.path.join(dir_path, filename)
    with open(file_path, 'w') as file:
        file.truncate(file_size)
    assert os.path.getsize(file_path) == file_size
```
just skimming over it would have sufficed to see what's going on.

Populating `files_to_create_with_sizes` explicitly (with literals) would be rather verbose, but would reveal the resulting structure almost visually:

```python
files_to_create_with_sizes = [
    ('' 'a.txt', 102),
    ('b' 'c.txt', 304),
    ('b' 'd.txt', 10),
    ('b/e' 'f.txt', 2000),
    ('g/h/i/j' 'k.txt', 10),
]